### ISSUE TITLE
tokenizer.get_vocab() unexpectedly slow

### ISSUE THREAD

**Issue by Mindful** [huggingface/tokenizers#483]
I was trying to find a bottleneck in my code today and ended up with a cprofile output that looked like this:
```
   Ordered by: internal time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     6831   72.330    0.011   72.330    0.011 {method 'get_vocab' of 'tokenizers.Tokenizer' objects}
      599    3.769    0.006   76.107    0.127 train_model.py:34(<listcomp>)
     1970    0.197    0.000    0.197    0.000 {built-in method marshal.loads}
  200/199    0.172    0.001    0.175    0.001 {built-in method _imp.create_dynamic}
      600    0.110    0.000    0.110    0.000 {method 'encode' of 'tokenizers.Tokenizer' objects}
        2    0.075    0.038    0.075    0.038 {built-in method from_file}
5773/5542    0.074    0.000    0.341    0.000 {built-in method builtins.__build_class__}
.....
```

Changing my code so that get_vocab() was only called once and then saved to a variable fixed the issue.
I assume that there is an unavoidable expensive computation somewhere, but I think having said computation attached to a getter method is potentially confusing. At the very least, this was very much not where I expected to find my bottleneck. 

My `tokenizers` version is 0.8.1rc2.

**Comment by n1t0** [comment]
Indeed, `get_vocab` is not meant to be called repeatedly. For such use-cases, you should probably use `token_to_id` and `id_to_token` methods that are generally better suited, and free to call. 

When calling `get_vocab` it actually allocates a new Python `dict` with the whole vocabulary. The original vocabulary lives on the Rust side as a `HashMap`, and so to make it available to Python, we have to convert it.

Can I ask why you need to get the vocabulary?

**Comment by Mindful** [comment]
> Can I ask why you need to get the vocabulary?

I didn't, I needed the ID for a specific token and just found `get_vocab` before `token_to_id` or `id_to_token`. I'll use the single token methods going forward, thanks for that.

I feel like in an ideal world there wouldn't be heavy operations attached to getters but I understand about the Rust bindings; I skimmed the code when I first saw this and came to the conclusion it was likely something in that vein. 
Not a huge issue either way, but maybe something to mention in the documentation somewhere?
